### PR TITLE
Features/post subcontact

### PR DESCRIPTION
**Pull request by gdchamal**
Add create and delete API for Contact related object Phone, Email and IM

Index update logic does not work anymore and I'll temporary disable it, by a commit in master of caliopen.base

### CODE REVIEW

File: caliopen/api/user/contact.py [hunk 1]
```diff
@@ -84,12 +97,48 @@ def __init__(self, request):
...
+    def _create(self, contact_id, params, add_func, return_obj):
+        """Create sub object from param using add_func."""
+        contact = CoreContact.get(self.user, contact_id)
+        created = getattr(contact, add_func)(params)
+        log.debug('Created object {} for contact {}'.
+                  format(created.address_id, contact.contact_id))
+        return return_obj.build(created).serialize()
+
+    def _delete(self, relation_id, delete_func):
+        """Delete sub object relation_id using delete_fund."""
```
Comment (iamdey) [inline caliopen/api/user/contact.py]: `delete_func` ?

**Comment by iamdey** [comment]
LGTM.

I may have question about im : does the type of im free ? we can accept anything 'facebook', 'face Book', 'nananananère' etc. ?

File: caliopen/api/user/contact.py (hunk 1 above)
Comment (gdchamal) [inline caliopen/api/user/contact.py]: the Contact method to delete a sub object (delete_im, delete_email, etc)

**Comment by gdchamal** [comment]
No types are not free, I don't set accepted value at this time, they can be found in base.user.parameters.contact

**Comment by iamdey** [comment]
thanks

**Comment by iamdey** [comment]
I don't see managing Phones but PostalAddress (cf. PR description)

**Comment by gdchamal** [comment]
Yes postaladdress and not phone at this time. Will push in another branch missing objects

**Pull request merged**